packages/runtime/src/index.js

```diff
@@ -29,15 +29,15 @@ export default ({
   const {code} = transform(jsx)
 
   const keys = Object.keys(fullScope)
-  const values = keys.map(key => fullScope[key])
+  const values = Object.values(fullScope)
   // eslint-disable-next-line no-new-func
   const fn = new Function(
     '_fn',
     'React',
     ...keys,
     `${code}
 
-  return React.createElement(MDXContent);`
+  return React.createElement(MDXContent, { components });`
```

Comment:
We need to manually pass components as a prop, otherwise `const { components = {} } = this.props` in render will override them.